app/javascript/components/widgets/weather/current-weather.jsx

```diff
@@ -30,17 +39,57 @@ const WeatherDescription = ({ baseDescription, weather }) => {
   return baseDescription;
 };
 
+export const AnticipatedPrecipitation = ({ weather }) => {
+  const precip = [];
+  for (let i = 0; i < weather.minutely.length; i += 1) {
+    if (weather.minutely[i].precipitation !== 0) {
+      precip.push([i, { value: weather.minutely[i].precipitation }]);
+    }
+  }
+  if (precip.length) {
+    let result = 0;
+    precip.forEach(currentValue => {
+      result += currentValue[1].value;
+    });
```

Comment:
Could be a nice refactor here into FP using `map`, `filter`, and `reduce`.

Reply:
Resolved (: